### DISCUSSION TITLE
Graphs with Inputs

### DISCUSSION THREAD

**Discussion by djwessel** [dagster-io/dagster#5430] [Q&A] [answered]
I have been trying out the new op/graph/job API and am really enjoying some of the changes. Right now I am playing around with nesting different graphs to allow for maximum reuse. For example, I have the following graph:
```Python
@graph
def reshape_and_convert(data, foo=10, bar=20):
    reshaped_data = reshape(data, foo=foo)
    converted_data = convert(reshaped_data, bar=bar)

    return converted_data
```
I can reuse this graph in different jobs/graphs, but if I want to test the functionality of the graph by itself, I run into the following issues.
1. I cannot directly call the graph directly `reshape_and_convert(test_data, foo=20)` as I get the following error: `dagster.core.errors.DagsterInvariantViolationError: Attempted to call graph 'reshape_and_convert' outside of a composition function. Invoking graphs is only valid in a function decorated with @job or @graph.`
2. I also cannot use the `execute_in_process()` function as I get `dagster.core.errors.DagsterInvariantViolationError: Graphs with inputs cannot be used with execute_in_process at this time.`

I have found the following hacky workaround, but I am not satisfied:
```Python
@op(config_schema={"input": Any})
def input_op(context):
    return context.op_config["input"]

@graph
def wrapped_reshape_and_convert():
    data = input_op()
    return reshape_and_convert(data)

result = wrapped_reshape_and_convert.execute_in_process(
    run_config={"ops": {"input_op": {"config": {"input": vector}}}}
)
```
So my questions are as follows:
1. Is there a better approach to testing graphs with inputs?
2. Given the error message "Graphs with inputs cannot be used with execute_in_process **at this time**." (emphasis mine) is there a plan to allow graphs with inputs to be used with `execute_in_process` in the future?

**Comment by sryza** [comment, marked as answer]
Hey @djwessel - there isn't currently a better approach to testing graphs with inputs, but, as you noticed, yes - in the future, we'd like `execute_in_process` to directly accept input values.  We don't yet have a date we plan to ship this, but will consider your post as a vote for prioritizing it.

**Reply by djwessel** [reply]
Thanks! I will keep an eye out for that in the future.

**Comment by doranmartinez** [comment]
Any update on this feature?
Seems related: https://github.com/dagster-io/dagster/discussions/14284
